src/js/utils/Util.js

```diff
@@ -92,37 +92,66 @@ const Util = {
   },
 
   /**
+   * Underscore.js 1.8.3. For more information,
+   * see http://underscorejs.org/docs/underscore.html#section-82
    * @param {Function} func A callback function to be called
    * @param {Number} wait How long to wait
-   * @returns {Function} A function, that, after triggered the first time, will
-   * wait for a period of time before the next trigger. If triggered during the
-   * wait, the function will be invoked immediately after the wait is over.
-   * This function is specifically made for React events, hence the nativeEvent
-   * lookup.
+   * @param {Object} [options]
+   * @param {Object} [options.leading = false]
+   * @param {Object} [options.trailing = true]
+   * @returns {Function} A function, that, when invoked, will only be triggered
+   * at most once during a given window of time. Normally, the throttled
+   * function will run as much as it can, without ever going more than once per
+   * wait duration; but if you’d like to disable the execution on the leading
+   * edge, pass {leading: false}. To disable execution on the trailing edge,
+   * ditto.
    */
-  throttleScroll(func, wait) {
-    let canCall = true;
-    let callWhenReady = false;
-
-    const resetCall = function () {
-      if (callWhenReady) {
-        setTimeout(resetCall.bind(this, arguments[0]), wait);
-        callWhenReady = false;
-        func.apply(this, arguments);
-      } else {
-        canCall = true;
+  throttle(func, wait, options) {
+    var context;
+    var args;
+    var result;
+    var timeout = null;
+    var previous = 0;
+    if (!options) {
+      options = {};
+    }
+
+    var later = function () {
+      previous = Date.now();
+      if (options.leading === false) {
+        previous = 0;
+      }
+      timeout = null;
+      result = func.apply(context, args);
+      if (!timeout) {
```

Comment:
Won't this always be `true` because we set `timeout = null;` just above? I see this is an implementation from `underscore` so I guess we could just leave as is.